### ISSUE TITLE
[ipfix_decode_trecord] warning: ipfix_get_template_ident() failed

### ISSUE THREAD

**Issue by cameronkerrnz** [cameronkerrnz/libipfix#2]

**Comment by cameronkerrnz** [comment]
I think its running into some underestimated storage management. (the Netscaler has a lot of templates; some of which are quite large, although I'm sure you could/will find larger).

```
TEMPLATE RECORD:
 template id:  260
 nfields:      24
  field01: ie=0.138, len=4 (observationPointId)
  field02: ie=0.144, len=4 (exportingProcessId)
  field03: ie=0.148, len=8 (flowId)
  field04: ie=5951.129, len=4 (reverse netscaler_transaction_id)
  field05: ie=5951.133, len=4 (reverse netscaler_connection_id)
  field06: ie=0.60, len=1 (ipVersion)
  field07: ie=0.4, len=1 (protocolIdentifier)
  field08: ie=0.210, len=2 (paddingOctets)
  field09: ie=0.27, len=16 (sourceIPv6Address)
  field10: ie=0.28, len=16 (destinationIPv6Address)
  field11: ie=0.7, len=2 (sourceTransportPort)
  field12: ie=0.11, len=2 (destinationTransportPort)
  field13: ie=0.2, len=8 (packetDeltaCount)
  field14: ie=0.1, len=8 (octetDeltaCount)
  field15: ie=0.6, len=1 (tcpControlBits)
  field16: ie=5951.132, len=8 (reverse netscaler_flow_flags)
  field17: ie=0.154, len=8 (flowStartMicroseconds)
  field18: ie=0.155, len=8 (flowEndMicroseconds)
  field19: ie=5951.128, len=4 (reverse netscaler_round_trip_time)
  field20: ie=0.14, len=4 (egressInterface)
  field21: ie=0.10, len=4 (ingressInterface)
  field22: ie=5951.151, len=4 (reverse netscaler_app_name_app_id)
  field23: ie=5951.192, len=16 (reverse netscaler_connection_chain_id)
  field24: ie=5951.193, len=1 (reverse netscaler_connection_chain_hop_count)
[ipfix_db] table ipfix_5 found
[ipfix_parse_msg] set1: 128 bytes decoded
[ipfix_parse_msg] set1: decode template, setlen=1240, left=504

Breakpoint 2, ipfix_get_template_ident (t=0x6315a0,
    identbuf=0x6315f8 "1_2_4_6_7_a_b_e_1b_1c_3c_8a_90_94_9a_9b_d2_v173f_81_82_83_84_85_8c_8d_8e_8f_97_9e_aa_ab_b7_b9_ba_be_c0_c1_cd_10b_10c_10", buflen=120) at ipfix_col.c:2010
2010            return -1;
```

So `len` will be 24 \* 10 = 240 bytes, which is enough; its buflen which is insufficient:

``
    if ( len >= buflen ) {
        return -1;
    }

```

GDB shows that it is 120 (and len is 120 at that point). The calling code:

```

(gdb) up
#1  0x00007ffff7de209e in ipfix_decode_trecord (s=0x62ec00, setid=2, buf=0x7ffffffee084 "\001\005", len=504, nread=0x7ffffffedc9c, templ=0x7ffffffedc90) at ipfix_col.c:760
760         if ( ipfix_get_template_ident( t, n->ident, sizeof(n->ident) )<0 ) {
(gdb) list
755             }
756         }
757
758         /* generate template ident (experimental)
759          */
760         if ( ipfix_get_template_ident( t, n->ident, sizeof(n->ident) )<0 ) {
761             mlogf( 1, "[%s] warning: ipfix_get_template_ident() failed\n", func );
762         }
763
764         /*\* insert template into tlist

```

The relevant structure definition:

```

typedef struct ipfixt_node
{
    struct ipfixt_node   _next;
    time_t               expire_time;
    ipfix_template_t     *ipfixt;
    char                 ident[MAXTEMPLIDENT];
#ifdef DBSUPPORT
    char                 tablename[MAXTABLENAMELEN+1]; /_ make this dynamic _/
    unsigned             template_id;      /_ id from template table (hack) _/
    unsigned             message_snr;      /_ last ipfix message snr (hack) */
#endif
} ipfixt_node_t;

```

So `MAXTEMPLIDENT` is too small.

```

#define MAXTEMPLIDENT        120

```

I'll double it, which makes it 240, which is the same as what `len` was further above (at least in that invocation)

```

**Comment by cameronkerrnz** [comment]
Resolved by d747dde040775048eae2e381cd6d6446093ce81d
